### PR TITLE
Allow configurable build platform

### PR DESCRIPTION
**Pull request by jordwalke**
Summary:Allows configuration of the build platform, which affects some
details such as path separators. It might affect more than that, but for
now it's just about path separators.

Test Plan:
Successfully built for windows!

Reviewers: @andreypopp 

I haven't used Flow in a long time, so not sure if I did something wrong but everything seems to work.

### CODE REVIEW

**Comment by jordwalke** [comment]
BTW: I merely implemented it this way to get the ball rolling. Happy to rearrange some stuff. For example we can make it so that it builds the cygwin equivalents for all the Makefiles and `build-env` etc.

File: esy/src/bin/esy.js [hunk 1]
```diff
@@ -258,12 +309,21 @@ async function buildCommand(sandboxPath) {
...
+async function buildEjectCommand(
+  sandboxPath,
+  _commandName,
+  _buildEjectPath,
+  buildPlatformArg,
+) {
+  const buildPlatform: BuildPlatform = determineBuildPlatformFromArgument(buildPlatformArg);
   const buildEject = require('../builders/makefile-builder');
   const sandbox = await getBuildSandbox(sandboxPath);
+  console.log(process.argv.join('--'));
```
Comment (andreypopp) [inline esy/src/bin/esy.js]: debug log

File: esy/src/types.js [hunk 2]
```diff
@@ -104,6 +114,9 @@ export type BuildConfig = {
 
   /**
    * Path to a sandbox root.
+   * TODO: Model this as sufficiently abstract to prevent treating this as a
+   * string containing a file path. It very well might contain the name of an
+   * environment variable that eventually will contain the actual path.
```
Comment (andreypopp) [inline esy/src/types.js:119]: Yep, actually this is a make variable in case of `makefile-builder`.

**Review by andreypopp — APPROVED** [review]
Looks good to me!

**Pull request merged**